addons/stock/models/stock_location.py

```diff
@@ -184,6 +207,56 @@ def should_bypass_reservation(self):
         self.ensure_one()
         return self.usage in ('supplier', 'customer', 'inventory', 'production') or self.scrap_location or (self.usage == 'transit' and not self.company_id)
 
+    def _check_can_be_used(self, product, quantity=0, package=None):
+        """Check if product/package can be stored in the location. Quantity
+        should in the default uom of product, it's only used when no package is
+        specified."""
+        self.ensure_one()
+        if package and package.package_type_id:
+            return self._check_package_storage(product, package=package)
+        return self._check_product_storage(product, quantity)
+
+    def _check_product_storage(self, product, quantity):
+        """Check if a number of product can be stored in the location. Quantity
+        should in the default uom of product."""
+        self.ensure_one()
+        if self.storage_category_id:
+            # check weight
+            if self.storage_category_id.max_weight < product.weight:
+                return False
+            # check if only allow new product when empty
+            if self.storage_category_id.allow_new_product == "empty" and any(not float_is_zero(q.quantity, precision_rounding=q.product_id.uom_id.rounding) for q in self.quant_ids):
+                return False
+            # check if only allow same product
+            if self.storage_category_id.allow_new_product == "same" and self.quant_ids and self.quant_ids.product_id != product:
+                return False
+            # check if enough space
+            product_capacity = self.storage_category_id.product_capacity_ids.filtered(lambda pc: pc.product_id == product)
+            if product_capacity and quantity > product_capacity.quantity:
```

Comment:
capacity should be <= current capacity + new product quantity

Reply:
Actually when I call `_check_can_be_used`, I already set quantity = current capacity + new product quantity
https://github.com/odoo/odoo/blob/c4a141a472bd7dbd2bce06ed3d5ee0a2eadab136/addons/stock/models/product_strategy.py#L120
But I forgot to change here
https://github.com/odoo/odoo/blob/c4a141a472bd7dbd2bce06ed3d5ee0a2eadab136/addons/stock/models/product_strategy.py#L104
I will change it